### PR TITLE
14066 - Remove Unused Image dialog improvements - Layout and file sizes

### PR DESCRIPTION
**Pull request by BrentEaston**

### CODE REVIEW

File: vassal-app/src/main/java/VASSAL/tools/DataArchive.java [hunk 1]
```diff
@@ -169,6 +170,26 @@ public InputStream getImageInputStream(String fileName)
       "'" + imageDir + fileName + "' not found in " + getName());
   }
 
+  /**
+   * Return the size in bytes of a file in the Archive
+   * @param fileName File name (include path if required)
+   * @return size in bytes
+   */
+  public long getFileSize(String fileName) {
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/tools/DataArchive.java:178]: Is there a reason you're doing this instead of using `FileArchive.getSize()`?

Comment (BrentEaston) [reply vassal-app/src/main/java/VASSAL/tools/DataArchive.java:178]: Just unfamiliarity with the code. Will FileArchive,getSize() return the compressed or uncompressed size? Is there an easy way to get the compressed size? That would be better as it relates directly to the issue of reducing the module size.

Comment (uckelman) [reply vassal-app/src/main/java/VASSAL/tools/DataArchive.java:178]: It's uncompressed size---but most images barely compress at all, so uncompressed size is a pretty good proxy for compressed size.

Comment (BrentEaston) [reply vassal-app/src/main/java/VASSAL/tools/DataArchive.java]: Turned out to be easy to access the compressed size. Try this new version

**Comment by uckelman** [comment]
This looks ok. Have you tried it when editing an extension?

File: vassal-app/src/main/java/VASSAL/tools/DataArchive.java [hunk 2]
```diff
@@ -202,7 +202,7 @@ public InputStream getInputStream(String fileName)
     // HFS+ filesystem and the filename contains decomposable characters,
     // so it got munged into NFD. Aauugh! Seriously, DIAF Apple!
     final String nfd = Normalizer.normalize(fileName, Normalizer.Form.NFD);
-    if (fileName != nfd) {
+    if (!fileName.equals(nfd)) {
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/tools/DataArchive.java:205]: Good catch.

**Comment by BrentEaston** [comment]
Ah, good point. Looks like a couple of problems editing extensions. 
1. Images from the module are being included in the list as well as images from the extension
2. All images from the extension are reporting a size of -1 byte (the fall back if ZipArchive.getCompressedSized() can't find the file.

**Comment by BrentEaston** [comment]
That last push should resolve both issues and add support for removing images from Extensions.

**Pull request merged**